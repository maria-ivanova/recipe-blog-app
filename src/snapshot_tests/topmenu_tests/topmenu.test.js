import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Routing from '../helpers/router.js';
import Topmenu from '../../components/topmenu.js';

describe('Topmenu Component', () => {
    it('should render layout without categories', () => {
        const tree = renderer.create(
            <Routing>
                <Topmenu />
            </Routing>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should render layout with categories', () => {
        const wrapper = mount(
            <Routing>
                <Topmenu />
            </Routing>
        )

        let component = wrapper.find('Topmenu');
        component.setState({links: ['test category']});
        component = component.update();

        expect(component.find('.topmenu a').length).toBe(2);
    })
})