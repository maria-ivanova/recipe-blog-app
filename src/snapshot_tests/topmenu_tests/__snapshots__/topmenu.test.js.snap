// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topmenu Component should render layout without categories 1`] = `
<section
  className="sec nav_sec"
>
  <div
    className="container"
  >
    <nav
      className="topmenu"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Начало
      </a>
    </nav>
  </div>
</section>
`;
